azure-pipelines.yml

```diff
@@ -149,9 +155,46 @@ jobs:
       Expand-Archive "openblas.zip" $tmpdir
       $pyversion = python -c "from __future__ import print_function; import sys; print(sys.version.split()[0])"
       Write-Host "Python Version: $pyversion"
-      $target = "C:\\hostedtoolcache\\windows\\Python\\$pyversion\\$(PYTHON_ARCH)\\lib\\openblas.a"
+      $target_include = Resolve-Path "C:\\hostedtoolcache\\windows\\Python\\$pyversion\\$(PYTHON_ARCH)\\include\\" | Select -ExpandProperty Path
+      $target_static = "C:\\hostedtoolcache\\windows\\Python\\$pyversion\\$(PYTHON_ARCH)\\lib\\openblas.a"
+      $target_lib = "C:\\hostedtoolcache\\windows\\Python\\$pyversion\\$(PYTHON_ARCH)\\lib\\openblas.lib"
+      $target_dll = "C:\\hostedtoolcache\\windows\\Python\\$pyversion\\$(PYTHON_ARCH)\\lib\\openblas.dll"
       Write-Host "target path: $target"
-      cp $tmpdir\$(BITS)\lib\libopenblas_v0.3.3-186-g701ea883-gcc_7_1_0.a $target
+      # we now use DLL (dynamic) libs intead of static on Windows
+      # put dumpbin.exe on our PATH
+      $TOOLSPATH = Resolve-Path "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.*\bin\Host$(PYTHON_ARCH)\$(PYTHON_ARCH)" | Select -ExpandProperty Path
+      echo $env:PATH
+      $env:PATH = $env:PATH + ";" +  $TOOLSPATH
+      $src_include = Resolve-Path $tmpdir\$(BITS)\include\*
+      Copy-Item $src_include -Destination $target_include -Recurse
+      ls $src_include
+      ls $target_include
+      cp $tmpdir\$(BITS)\lib\libopenblas_v0.3.3-186-g701ea883-gcc_7_1_0.a $target_static
+      cp $tmpdir\$(BITS)\bin\libopenblas_v0.3.3-186-g701ea883-gcc_7_1_0.dll $target_dll
+      # we need to generate an import library (LIB) for the DLL
+      # this should probably eventually happen upstream in MacPython openblas
+      # wheel builds, I suspect
+      # see: https://stackoverflow.com/a/9946390/2942522
+      dumpbin /exports $target_dll > exports.txt
+      echo LIBRARY OPENBLAS > openblas.def
+      echo EXPORTS >> openblas.def
+
+      $linecounter = 0
+      foreach($line in [System.IO.File]::ReadLines("./exports.txt"))
+      {
+         $linecounter++
+         If ($linecounter -gt 19) {
+            $items = $line.split()[0..3] 
+            echo $items >> openblas.def
+         }
+      }
+      If ($(BITS) -eq 32) {
+         lib /def:openblas.def /out:openblas.lib /machine:x86
+      }
+      If ($(BITS) -eq 64) {
+         lib /def:openblas.def /out:openblas.lib /machine:x64
```

Comment:
This should probably happen in MacPython wheel builds, I think, if it is truly needed. Certainly the lack of a `lib` file was holding back the builds before I did this.